This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit 314e19c1afe2cc0fd61588888df358ad56003f36
Author: Neal H. Walfield <neal@sequoia-pgp.org>
Date:   Fri Feb 23 14:30:06 2024 +0100

    Upgrade sequoia-cert-store and sequoia-wot.
    
      - Upgrade `sequoia-cert-store` and `sequoia-wot` and adapt to their
        API changes.

Index: sequoia-sq/src/commands/cert/import.rs
===================================================================
--- sequoia-sq.orig/src/commands/cert/import.rs
+++ sequoia-sq/src/commands/cert/import.rs
@@ -46,7 +46,7 @@ pub fn dispatch<'store>(mut config: Conf
                     Ok(raw_cert) => LazyCert::from(raw_cert),
                     Err(err) => {
                         wprintln!("Error parsing input: {}", err);
-                        stats.errors += 1;
+                        stats.inc_errors();
                         continue;
                     }
                 };
@@ -57,7 +57,7 @@ pub fn dispatch<'store>(mut config: Conf
                 if let Err(err) = cert_store.update_by(Arc::new(cert), &mut stats) {
                     wprintln!("Error importing {}, {:?}: {}",
                               fingerprint, userid, err);
-                    stats.errors += 1;
+                    stats.inc_errors();
                     continue;
                 } else {
                     wprintln!("Imported {}, {}", fingerprint, Safe(&userid));
@@ -72,7 +72,8 @@ pub fn dispatch<'store>(mut config: Conf
 
     wprintln!("Imported {} new certificates, updated {} certificates, \
                {} certificates unchanged, {} errors.",
-              stats.new, stats.updated, stats.unchanged, stats.errors);
+              stats.new_certs(), stats.updated_certs(),
+              stats.unchanged_certs(), stats.errors());
 
     Ok(result?)
 }
Index: sequoia-sq/src/commands/net.rs
===================================================================
--- sequoia-sq.orig/src/commands/net.rs
+++ sequoia-sq/src/commands/net.rs
@@ -113,10 +113,10 @@ pub fn import_certs(config: &mut Config,
     }
 
     wprintln!("\nImported {}, updated {}, {} unchanged, {}.",
-              stats.new.of("new certificate"),
-              stats.updated.of("certificate"),
-              stats.unchanged.of("certificate"),
-              stats.errors.of("error"));
+              stats.new_certs().of("new certificate"),
+              stats.updated_certs().of("certificate"),
+              stats.unchanged_certs().of("certificate"),
+              stats.errors().of("error"));
 
     wprintln!("\nAfter checking that a certificate really belongs to the \
                stated owner, you can mark the certificate as authenticated \
Index: sequoia-sq/src/commands/pki/link.rs
===================================================================
--- sequoia-sq.orig/src/commands/pki/link.rs
+++ sequoia-sq/src/commands/pki/link.rs
@@ -765,7 +765,7 @@ pub fn retract(mut config: Config, c: li
 pub fn list(mut config: Config, c: link::ListCommand)
     -> Result<()>
 {
-    let mut cert_store = config.cert_store_or_else()?;
+    let cert_store = config.cert_store_or_else()?;
     cert_store.prefetch_all();
 
     let trust_root = config.local_trust_root()?;
Index: sequoia-sq/src/sq.rs
===================================================================
--- sequoia-sq.orig/src/sq.rs
+++ sequoia-sq/src/sq.rs
@@ -423,7 +423,7 @@ impl<'store> Config<'store> {
             cert_store::CertStore::empty()
         };
 
-        let mut keyring = cert_store::store::Certs::empty();
+        let keyring = cert_store::store::Certs::empty();
         let mut error = None;
         for filename in self.keyrings.iter() {
             let f = std::fs::File::open(filename)
Index: sequoia-sq/Cargo.toml
===================================================================
--- sequoia-sq.orig/Cargo.toml
+++ sequoia-sq/Cargo.toml
@@ -94,7 +94,7 @@ version = "0.25"
 default-features = false
 
 [dependencies.sequoia-cert-store]
-version = "0.4"
+version = "0.5"
 
 [dependencies.sequoia-net]
 version = "0.28"
@@ -109,7 +109,7 @@ default-features = false
 version = "0.6"
 
 [dependencies.sequoia-wot]
-version = "0.9"
+version = "0.10"
 
 [dependencies.serde]
 version = "1.0.137"
